action/protocol/execution/evm/evm.go

```diff
@@ -418,6 +421,18 @@ func executeInEVM(ctx context.Context, evmParams *Params, stateDB *StateDBAdapte
 		// After EIP-3529: refunds are capped to gasUsed / 5
 		refund = (evmParams.gas - remainingGas) / params.RefundQuotientEIP3529
 	}
+	// adjust refund due to dynamicGas
+	var (
+		refundBeforeSnapshot = evm.RefundBeforeSnapshot
+		currentRefund        = stateDB.GetRefund()
+	)
+	if !g.IsOkhotsk(blockHeight) && refundBeforeSnapshot > 0 && refundBeforeSnapshot != currentRefund {
+		if refundBeforeSnapshot > currentRefund {
```

Comment:
add one more comment before L430?
`// stateDB.refund = refundBeforeSnapshot`

Reply:
comment is on L424